airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/data/JsonToAirbyteValue.kt

```diff
@@ -0,0 +1,154 @@
+                            )
+                    convert(json, option)
+                }
+                is UnknownType -> UnknownValue("From $schema: $json")
+            }
+        } catch (t: Throwable) {
+            return UnknownValue(t.message ?: "Unknown error")
+        }
+    }
+
+    private fun matchesStrict(schema: AirbyteType, json: JsonNode): Boolean {
+        when (schema) {
+            is ArrayType -> return json.isArray
+            is ArrayTypeWithoutSchema -> return json.isArray
+            is BooleanType -> return json.isBoolean
+            is DateType -> return json.isTextual
+            is FieldType -> return matchesStrict(schema.type, json)
+            is IntegerType -> return json.isNumber
+            is NullType -> return json.isNull
+            is NumberType -> return json.isNumber
+            is ObjectType -> return json.isObject
+            is ObjectTypeWithoutSchema -> return json.isObject
+            is ObjectTypeWithEmptySchema -> return json.isObject
+            is StringType -> return json.isTextual
+            is TimeType -> return json.isTextual
+            is TimestampType -> return json.isTextual
+            is UnionType -> return schema.options.any { matchesStrict(it, json) }
+            is UnknownType -> return false
+        }
+    }
+
+    private fun matchesPermissive(schema: AirbyteType, json: JsonNode): Boolean {
+        return try {
+            convert(json, schema) !is UnknownValue
+        } catch (t: Throwable) {
+            false
+        }
+    }
+
+    private fun toBoolean(json: JsonNode): BooleanValue {
+        return when {
+            json.isBoolean -> BooleanValue(json.asBoolean())
+            json.isIntegralNumber -> BooleanValue(json.asLong() != 0L)
+            json.isFloatingPointNumber -> BooleanValue(json.asDouble() != 0.0)
+            json.isTextual -> BooleanValue(json.asText().toBooleanStrict())
+            else -> throw IllegalArgumentException("Could not convert $json to Boolean")
+        }
+    }
+
+    private fun toInteger(json: JsonNode): Long {
+        return when {
+            json.isBoolean -> if (json.asBoolean()) 1L else 0L
+            json.isIntegralNumber -> json.asLong()
+            json.isFloatingPointNumber -> json.asDouble().toLong()
+            json.isTextual -> json.asText().toLong()
+            else -> throw IllegalArgumentException("Could not convert $json to Integer")
+        }
+    }
+
+    private fun toNumber(json: JsonNode): NumberValue {
+        return when {
+            json.isBoolean -> NumberValue(if (json.asBoolean()) 1.0 else 0.0)
+            json.isIntegralNumber -> NumberValue(json.asLong().toDouble())
+            json.isFloatingPointNumber -> NumberValue(json.asDouble())
+            json.isTextual -> NumberValue(json.asText().toDouble())
+            else -> throw IllegalArgumentException("Could not convert $json to Number")
+        }
+    }
+
+    private fun fromAny(value: Any?): AirbyteValue {
```

Comment:
I don't think I understand this function here - this class is all about converting JsonNode -> AirbyteValue, we should never be handling raw Any? (there's a few calls where it looks like we're passing a JsonNode to fromAny, which sounds wrong)

Reply:
yeah, I fixed that